### ISSUE TITLE
a11y: Fix keyboard navigation, label and focus display issues on the import yaml screen

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12778]
a11y Issues: 123, 125, 124, 128

Fix keyboard navigation and focus issues on the import yaml screen:

**Error message and tooltip**

The tooltip content should be available when the "Error" icon receives keyboard focus.

![image](https://github.com/user-attachments/assets/52fdb191-5ec7-4b84-b2b1-75f910ac1d95)

**Text Area**

Keyboard focus should be available for all interactive elements on page without any trap of focus.

![image](https://github.com/user-attachments/assets/3561a27e-8720-4b98-9fb2-47a9e5864303)

**Issue 124 - Missing textual description for informative image**

The informative image lacks textual description that resulted in screen reader users unable to access the information conveyed through the image. Additionally, image is defined through CSS background property.

Location: Error icon in the modal dialog:

![Image](https://github.com/user-attachments/assets/7775cd57-e475-474a-bd31-13328d76588f)

**Issue 128 - Error messages difficult to locate**

When users activate "Import" button after uploading file using "Read from File" button, the mentioned error message gets dynamically displayed and user's focus remains on that button resulted in screen reader users finding it difficult to find the errors occurred in the component.

![image](https://github.com/user-attachments/assets/0ae4d8b4-f116-4c87-8ab6-b2915cabeecc)

Apply the following changes:
• Specify role="alert" to ensure that the error message is announced to screen reader users. 
• Set focus on textarea using the JavaScript .focus() method as soon as the "Import" button is activated.

**Comment by aalves08** [comment]
To add some notes for the future:

First thing to handle is that the `CodeMirror` itself is a focus-trap, which should not happen. [Codemirror v6](https://codemirror.net/examples/tab/) has a handler for this that our current [v5](https://codemirror.net/5/doc/manual.html) doesn't seem to have. 

Proper way to handle this might just be consider the transition to v6 although it's not something really straightfoward as v6 is quite different from v5

**Comment by izaac** [comment]
@aalves08 I've noticed your codemirror comment. So the focus is trapped. I see one of the mitigations is to use `ESC` or backwards `TAB` to reach the other elements like `Import` and `Cancel`

Is this limitation captured in a backlog issue or it isn't something that's going to be improved in the future?

**Comment by aalves08** [comment]
@izaac , `CodeMirror` per default is a focus trap, since the `tab` key performs tabulation **inside** the code editor, hence why you can click `tab` 1000 times and never escape the focus trap.

Right now we are using CodeMirror `v5`, which doesn't have a way out of that focus trap.

There's CodeMirror `v6`, which does provide an out-of-the-box solution for this using the `escape` key, as per https://codemirror.net/examples/tab/

But updating from `v5` to `v6` is massive undertaking (has breaking changes and a big potencial for regressions), so we decided to try and do the same solution for `v5` , hence the behaviour you see today.

If you are focused on the code editor, you'll need to press `escape` to shift the focus to the outer container, then you can do `tab` or `shift-tab` to you pleasing.

We can create an issue to update CodeMirror in our codebase to `v6` but the UX of the solution is the same... For sure this is not a blocking situation, but if the idea is better and if there are no technical impediments, there might be room for improvements to this behaviour.

Since it's a pattern we think works just fine, we've also implemented the same solution for the `kubectl shell` (there's an action in the header for that), which is also, by default, a focus trap.

**Comment by izaac** [comment]
Rancher version: `v2.11-41098eee7331faf5d4a478dfcf66232867ed6a6f-head`
UI Version: `master 95c0062`
Firefox version: `135.0.1`
Chrome version: `133.0.6943.141`

Not a blocker but I wanted to make this clear for future reference. All validations described covered and required in the issue passed ✅
